### DISCUSSION TITLE
Discord Authentication, without email

### DISCUSSION THREAD

**Discussion by Vinnie64** [requarks/wiki#7286] [Help / Questions]
Is it possible for me to allow authentication into my wiki via discord oAuth without grabbing user emails? I do not plan on using them, and would like to avoid collecting them if possible.

**Comment by baviereteam** [comment]
I am also very interested about this. Some members do not wish to have their Discord account email displayed / added to the wiki because it could expose a part of their identity that is not known to their fellow server members.

For my personal needs I forked wiki and edited `/server/modules/authentication/discord/authentication.js` :
* Removed `email` from the scopes list on line 18
* Added a fake email entry in the profile object line 28 (`email: `discord-user-${profile.id}@localhost``)

But it would be way nicer if this could be supported natively.


It's also worth noting that Discord will add the permissions that the users granted to your app in the past to your scopes, so even if you only request `identify` but a user has accepted the app with the `email` permission in the past, you'll receive the `email` scope. Apparently the only way to stop that is if the user unauthorizes the app from their account (they'll get an authorize page next time they connect to it) which might not be do-able if a lot of people use your app.

**Reply by Vinnie64** [reply]
I love you.
Thank you.
